File: python/01_Two_Sum_jc.py
class Solution(object):
    def twoSum(self, nums, target):
        map = dict()
        result = [0,0]
        for i in range(len(nums)):
            if(map.has_key(nums[i])):
                result[0] = map.get(nums[i],None)
                result[1] = i
                break
            else:
                map[target - nums[i]] = i;
        return result
        
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        

File: c/01_Two_Sum_jc.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target) {
    int i,j;
    for(i = 0; i < numsSize; i++){
        for(j = i + 1; j < numsSize; j++){
            if(nums[i] + nums[j] == target){
                int* a;
                a = malloc(sizeof(int) * 2);
                a[0] = i;
                a[1] = j;
                return a;  
            }
        }
    }
    return NULL;
}